src/main/java/de/filefighter/rest/domain/filesystem/business/FileSystemBusinessService.java

```diff
@@ -98,14 +93,151 @@ public FileSystemItem getFileSystemItemInfo(long fsItemId, User authenticatedUse
+
+                    // save changes of parentFolder to db.
+                    Query query = new Query().addCriteria(Criteria.where("fileSystemId").is(parentFileSystemEntity.getFileSystemId()));
+                    Update newUpdate = new Update()
+                            .set("itemIds", parentFileSystemEntity.getItemIds())
+                            .set("visibleForUserIds", parentFileSystemEntity.getVisibleForUserIds())
+                            .set("visibleForGroupIds", parentFileSystemEntity.getVisibleForGroupIds())
+                            .set("editableForUserIds", parentFileSystemEntity.getEditableForUserIds())
+                            .set("editableForGroupIds", parentFileSystemEntity.getEditableFoGroupIds());
+                    mongoTemplate.findAndModify(query, newUpdate, FileSystemEntity.class);
+                } else {
+                    // No FileSystemEntities left in folder. -> can be deleted.
+                    fileSystemRepository.delete(parentFileSystemEntity);
+                    everythingWasDeleted = true;
+                }
+                fileSystemRepository.deleteAll(entitiesToBeDeleted);
+            }
         }
-        return pathToFind;
+        return everythingWasDeleted;
+    }
+
+    // ---------------- HELPER -------------------
+
+    public FileSystemEntity sumUpAllPermissionsOfFileSystemEntities(FileSystemEntity parentFileSystemEntity, List<FileSystemEntity> fileSystemEntities) {
+        HashSet<Long> visibleForUserIds = new HashSet<>();
+        HashSet<Long> visibleForGroupIds = new HashSet<>();
+        HashSet<Long> editableForUserIds = new HashSet<>();
+        HashSet<Long> editableGroupIds = new HashSet<>();
+
+        for (FileSystemEntity entity : fileSystemEntities) {
+            addPermissionsToSets(visibleForUserIds, visibleForGroupIds, editableForUserIds, editableGroupIds, entity);
+        }
+
+        parentFileSystemEntity.setVisibleForUserIds(Arrays.stream(visibleForUserIds.toArray(new Long[0])).mapToLong(Long::longValue).toArray());
+        parentFileSystemEntity.setVisibleForGroupIds(Arrays.stream(visibleForGroupIds.toArray(new Long[0])).mapToLong(Long::longValue).toArray());
+        parentFileSystemEntity.setEditableForUserIds(Arrays.stream(editableForUserIds.toArray(new Long[0])).mapToLong(Long::longValue).toArray());
+        parentFileSystemEntity.setEditableFoGroupIds(Arrays.stream(editableGroupIds.toArray(new Long[0])).mapToLong(Long::longValue).toArray());
+        return parentFileSystemEntity;
+    }
+
+    public void addPermissionsToSets(Set<Long> visibleForUserIds, Set<Long> visibleForGroupIds, Set<Long> editableForUserIds, Set<Long> editableGroupIds, FileSystemEntity fileSystemEntity) {
+        for (long i : fileSystemEntity.getVisibleForUserIds()) {
+            visibleForUserIds.add(i);
+        }
+        for (long i : fileSystemEntity.getVisibleForGroupIds()) {
```

Comment:
Set to list is also good supported by the language

Reply:
Maybe im to stupid. But for me nothing else worked.